common/src/main/java/com/microsoft/identity/common/internal/broker/MicrosoftAuthClient.java

```diff
@@ -83,7 +86,17 @@ public MicrosoftAuthClient(@NonNull final Context context,
 
             case MSAL_GET_INTENT_FOR_INTERACTIVE_REQUEST:
                 final Intent intent = microsoftAuthService.getIntentForInteractiveRequest();
-                return intent.getExtras();
+                final Bundle bundle = intent.getExtras();
+
+                //older brokers (pre-ContentProvider) are ONLY sending these values in the intent itself.
+                final String packageName = intent.getPackage();
+                final String className = intent.getComponent().getClassName();
+                if (!TextUtils.isEmpty(packageName) && !TextUtils.isEmpty(className)){
+                    bundle.putString(BROKER_PACKAGE_NAME, packageName);
+                    bundle.putString(BROKER_ACTIVITY_NAME, className);
+                }
```

Comment:
do we expect the else case to hit ever; if so should we log something to help in debugging?

Reply:
The else case happens when MSAL talks to the post-ContentProvider (newer) broker. In such case, everything is already packed inside the bundle.